themes/default/content/docs/intro/concepts/inputs-outputs.md

```diff
@@ -651,6 +651,327 @@ variables:
+
+The following example allows you to natively represent the definition of an AWS Step Function State Machine and easily embed outputs from other resources in your stack.
+
+```python
+state_machine = aws_native.stepfunctions.StateMachine("stateMachine",
+    role_arn=sfn_role.arn,
+    state_machine_type="EXPRESS",
+    definition_string=pulumi.Output.json_dumps({
+        "Comment": "A Hello World example of the Amazon States Language using two AWS Lambda Functions",
+        "StartAt": "Hello",
+        "States": {
+            "Hello": {
+                "Type": "Task",
+                "Resource": hello_function.arn,
+                "Next": "World",
+            },
+            "World": {
+                "Type": "Task",
+                "Resource": world_function.arn,
+                "End": True,
+            },
+        },
+    })
+)
+```
+
+### Example Parsing JSON Outputs to Objects
+
+The following is an example of Pulumi code that operates on a JSON object and removes all of the policy statements by replacing it with an empty list.
+
+```python
+json_IAM_policy = pulumi.Output.from_input('''
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "VisualEditor0",
+            "Effect": "Allow",
+            "Action": [
+                "s3:ListAllMyBuckets",
+                "s3:GetBucketLocation"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Sid": "VisualEditor1",
+            "Effect": "Allow",
+            "Action": "s3:*",
+            "Resource": "arn:aws:s3:::my-bucket"
+        }
+    ]
+}
+''')
+
+def update_policy(policy):
+    # delete the policy statements
+    policy.update({'Statement': []})
+    return policy
+
+policy_with_no_statements = pulumi.Output.json_loads(json_IAM_policy).apply(update_policy)
+```
+
+For more details view the Python documentation.
+
+{{% /choosable %}}
+{{% choosable language go %}}
+
+### Example Converting Outputs to JSON
+
+{{% notes type="info" %}}
+The Pulumi Go SDK does not currently support Marshalling and Unmarshalling Maps with unknowns.
+
+This is being tracked [here](https://github.com/pulumi/pulumi/issues/12460).
+{{% /notes %}}
+
+The following is an example of JSON Serializing pulumi Outputs in Go.
+
+```go
+pulumi.JSONMarshal(pulumi.ToMapOutput(map[string]pulumi.Output{
+    "bool": pulumi.ToOutput(true),
+    "int":  pulumi.ToOutput(1),
+    "str":  pulumi.ToOutput("hello"),
+    "arr": pulumi.ToArrayOutput([]pulumi.Output{
+        pulumi.ToOutput(false),
+        pulumi.ToOutput(1.0),
+        pulumi.ToOutput(""),
+        pulumi.ToMapOutput(map[string]pulumi.Output{
+            "key": pulumi.ToOutput("value"),
+        }),
+    }),
+    "map": pulumi.ToMapOutput(map[string]pulumi.Output{
+        "key": pulumi.ToOutput("value"),
+    }),
+    // currently unsupported as myResource is unknown
+    "unknown": myResource.ApplyT(func(res interface{}) (interface{}, error) { return "Hello World!", nil }),
+}))
+```
```

Comment:
I feel that this example could be more practical as I think people only work with `outputs` because the values are unknown.

Reply:
I ran this example on my machine and it worked when the "unknown" key was commented out.